src/zitadel.rs

```diff
@@ -20,11 +20,17 @@ const FAMEDLY_NAMESPACE: Uuid = uuid!("d9979cff-abee-4666-bc88-1ec45a843fb8");
 const FAMEDLY_USER_ROLE: &str = "User";
 
 /// A very high-level Zitadel client
+#[derive(Clone)]
 pub(crate) struct Zitadel {
+	/// ldap-sync configuration
+	zitadel_config: FamedlyConfig,
+	/// Optional set of features
+	feature_flags: FeatureFlags,
 	/// The backing Zitadel client
 	client: ZitadelClient,
-	/// ldap-sync configuration
-	config: Config,
+	/// LDAP attributes mapping for converting LDAP users to Zitadel users
+	// TODO: Don't like this here, but it's required for User::try_from_search_entry for now
+	attributes_mapping: Option<LdapAttributesMapping>,
```

Comment:
I agree that having this here is not ideal, it feels a bit counter to the purpose of the nice abstractions this PR otherwise introduces. Could we pass this information along the `SearchEntry` or some kind of wrapper type around it instead?

Reply:
The wrapper type already exists in the form of `User`, I don't think this module should be handling a `SearchEntry` at all anymore. Then `delete_users_by_email` could just be folded into the generic `delete_users` as well, and we wouldn't need multiple paths for things.

Reply:
Yes, this is pending. The problem is that the `User` is too much tangled with LDAP. The way to go is to refactor it to make a clear distinction between the Zitadel part and the LDAP part.